client/state/selectors/is-woocommerce-core-profiler-flow.ts

```diff
@@ -11,10 +11,22 @@ import type { AppState } from 'calypso/types';
  * @returns {?boolean}        Whether the user reached Calypso via the WooCommerce Core Profiler flow
  */
 export const isWooCommerceCoreProfilerFlow = ( state: AppState ): boolean => {
-	const allowedFrom = [ 'woocommerce-core-profiler' ];
+	const isCoreProfiler =
+		get( getInitialQueryArguments( state ), 'from' ) === 'woocommerce-core-profiler' ||
+		get( getCurrentQueryArguments( state ), 'from' ) === 'woocommerce-core-profiler';
+
+	const allowedPluginNames = [ 'woocommerce-payments' ];
+	const isPluginNameAllowed =
+		allowedPluginNames.includes(
+			get( getInitialQueryArguments( state ), 'plugin_name' ) as string
+		) ||
+		allowedPluginNames.includes(
+			get( getCurrentQueryArguments( state ), 'plugin_name' ) as string
+		);
+
 	return (
-		allowedFrom.includes( get( getInitialQueryArguments( state ), 'from' ) as string ) ||
-		allowedFrom.includes( get( getCurrentQueryArguments( state ), 'from' ) as string ) ||
+		isCoreProfiler ||
```

Comment:
Should this be `&&` ? If I'm understanding the scope correctly, we're only showing this if
1. the user has just completed the core-profiler and is redirected to wpcom for signup/login
2. the user has woocommerce-payments selected

Reply:
@rjchow The change is for the WooCommerce Payments onboarding process, where we don’t have `from=woocommerce-core-profiler`, but instead have `plugin_name=woocommerce-payments` (this will be added on the WC Payments side soon)

Reply:
If that's the case then I think it should have its own selector as `isWooCommerceCoreProfilerFlow` returning true when it's not actually a core profiler flow is going to be confusing?

Not trying to be nitpicky here, it's just that this whole login/sign-up flow is already a jumble of different flows and it it won't help us if we have misleading function names

Reply:
@rjchow No, you're not nitpicking :) I totally agree with you. It's just that I figured this is a good enough temporary solution until we have a project to clean up the connection page codes. 

Let's put this on hold for now. I'll work on someone else (hopefully, better than this 😄 ) this week and next week.